dali/dfuXRefLib/dfuxreflib.cpp

```diff
@@ -923,17 +964,60 @@ static bool parseFileName(const char *name,StringBuffer &mname,unsigned &num,uns
         name = nonrepdir.str();
     num = 0;
     max = 0;
+    unsigned dirPerPart = 0;
     for (;;) {
         char c=*name;
         if (!c)
             break;
+        if ((c=='d')&&(isdigit(name[1])))
+        {
+            // Checks that the prefix matches and numDevices > 1 to see if d[0-9]+ is a stripe number
+            mname.append(c);
+            name++;
+
+            Owned<IPropertyTreeIterator> planesIter = getPlanesIterator("data", nullptr);
+            ForEach(*planesIter)
+            {
+                const IPropertyTree &plane = planesIter->query();
+                if (startsWithIgnoreCase(mname.str(), plane.queryProp("@prefix")))
+                {
+                    if (plane.getPropInt("@numDevices") > 1)
+                    {
+                        while (*name&&isdigit(*name))
+                            name++;
+                        mname.append("$P$");
+                    }
+                    break;
+                }
+            }
+            c = *name;
+        }
+        if ((c=='/')&&(isdigit(name[1])))
+        {
+            mname.append(c);
+            name++;
+
+            // Get dir-per-part #
+            while (*name&&isdigit(*name))
+            {
+                dirPerPart = dirPerPart*10+(*name-'0');
+                name++;
+            }
+
+            c=*name;
+            mname.append("$P$");
+        }
         if ((c=='.')&&(name[1]=='_')) {
             unsigned pn = 0;
             const char *s = name+2;
             while (*s&&isdigit(*s)) {
                 pn = pn*10+(*s-'0');
                 s++;
             }
+
+            if (dirPerPart&&dirPerPart!=pn)
+                throw MakeStringException(-1, "dir-per-part # does not match part # of file");
```

Comment:
trivial: MakeStringException is deprecated in favour of makeStringException or makeStringExceptionV